core/src/clients/snyk/snyk_client.rs

```diff
@@ -0,0 +1,223 @@
+}
+
+#[tokio::test]
+async fn test_get_projects_from_org() {
+    let snyk_api = SnykRestClientImpl{};
+    let org_id = format!("f288c129-6c28-4b65-aec2-bd753e095b13");
+    let org_name= format!("Test Organization");
+    let snyk_token = get_snyk_access_token().await;
+    let good_request = snyk_api.get_projects_from_org(snyk_token.clone(), org_id.clone(), org_name.clone()).await;
+    println!("{:#?}", good_request);
+    // Validate we have a successfull response
+    assert!(good_request.is_ok(), "Response should be OK");
+
+    // Validate Option is not empty
+    assert!(good_request.as_ref().unwrap().is_some(), "Option in response should be SOME");
+
+    let org_projects = good_request.unwrap().clone().unwrap();
+
+    // Validate OrgProjects.projects field is not empty
+    assert!(!org_projects.projects.is_empty(), "The org should have a list of projects");
+
+    // Validate OrgProjects.org.id field
+    assert!(org_projects.org.id.is_some(), "Org ID should not be missing");
+    assert!(org_projects.org.id.unwrap() == org_id, "ID Should match what we used in the request");
+
+    // Validate OrgProjects.org.name field
+    assert!(org_projects.org.name.is_some(), "Org Name should not be missing");
+    assert!(org_projects.org.name.unwrap() == org_name, "Name Should match what we used in the request");
+
+    let bad_request = snyk_api.get_projects_from_org(snyk_token, format!("123-abc"), org_name.clone()).await;
+    assert!(bad_request.is_err(), "We made a bad request and this should produce an error");
+
+   
+
+}
+
+#[tokio::test]
+async fn test_get_projects_from_org_list() {
+    let snyk_api = SnykRestClientImpl{};
+    let org_list: Vec<Org> = vec![
+        Org{
+            id: Some(format!("230a9cf3-8880-43bb-a8ff-c69f372d4fc3")), 
+            name: Some(format!("PMPP [PMPP]"))
+        },
+        Org{
+            id: Some(format!("3957864e-d4a0-4290-bc83-750a2ea495e8")), 
+            name: Some(format!("MACFIN [MACFIN]"))
+        },
+        Org{
+            id: Some(format!("a1738e18-1b90-4ee9-b3dc-10a3b7c51953")), 
+            name: Some(format!("HIOS [HIOS]"))
+        }
+    ];
+    let snyk_token = get_snyk_access_token().await;
+    let get_proj_results = snyk_api.get_projects_from_org_list(snyk_token, org_list).await;
+    println!("{:#?}", get_proj_results);
+    // Validate we have a successfull response
+    assert!(!get_proj_results.0.is_empty(), "The first part of the tuple should have data in it");
+    assert!(get_proj_results.1.is_empty(), "The second part of the tuple should be empty (no errors found)");
+}
+
+#[tokio::test]
+async fn test_get_sbom_from_snyk() {
+    let snyk_api = SnykRestClientImpl{};
+    let snyk_token = get_snyk_access_token().await;
+    let result = snyk_api.get_sbom_from_snyk(snyk_token.clone(), format!("a1738e18-1b90-4ee9-b3dc-10a3b7c51953"), format!("b85a413d-b972-4271-9a78-87625c108c59")).await;
```

Comment:
Same comment about guids. I think the secret scanner will end up getting triggered by this.